### DISCUSSION TITLE
How to trigger container artwork from within a custom script

### DISCUSSION THREAD

**Discussion by ghlupe2** [gerbera/gerbera#1915] [Q&A]
Currently I am checking out the new artwork configuration based on certain metadata like %artist%, %genre% and, most recently (issue #1901), %composer%. It's working fine in both bubbleUPNP and web UI.

For composers I prefer a format like "Bach, Johann Sebastian (1685-1750)" and maintain an array of my top 100 classical composers on that behalf in import.js. Handling of multiple values for artist, genre, etc. as well requires scripting.

Is there any chance to trigger specific container resources from within a custom script?

**Comment by KarlStraussberger** [comment]
You can try to manipulate `chain.composer.res`. Take ids and tags from `metacontent_handler` and hope that `content_handler` doesn't overwrite it.

**Reply by ghlupe2** [reply]
That did not work for me, but pointed me to the right direction. I replaced `composer` by the auxdata tag `composersort` as title in `chain.composer `and put

`chain.composer.metaData[M_COMPOSER] = [ composersort ];`

My `%composer%.jpg` files are spelt exactly like the composer names in `composersort`. This worked well for Gerbera Web-UI. 
In bubbleUPNP and foobar2000 however, the composer jpg's didn't show up, maybe because `UPNP_CLASS_CONTAINER_MUSIC_COMPOSER` is not supported by them. 
It worked well after changing `upnpclass` to `UPNP_CLASS_CONTAINER_MUSIC_ARTIST` in `chain.composer`.
